content/en/blog/_posts/2023-04-11-volume-group-snapshot-alpha.md

```diff
@@ -0,0 +1,165 @@
+The creation and deletion of this object represents a user’s desire to create or delete a cluster resource (a group snapshot).
+
+VolumeGroupSnapshotContent
+* Created by the Snapshot Controller for a dynamically created VolumeGroupSnapshot. It contains information about the volume group snapshot including the volume group snapshot ID.
+This object represents a provisioned resource on the cluster (a group snapshot).
+The VolumeGroupSnapshotContent object binds to the VolumeGroupSnapshot for which it was created with a one-to-one mapping.
+
+VolumeGroupSnapshotClass
+* Created by cluster administrators to describe how volume group snapshots should be created. including the driver information, the deletion policy, etc.
+
+The Volume Group Snapshot objects are defined as CustomResourceDefinitions (CRDs). These CRDs must be installed in a Kubernetes cluster for a CSI Driver to support volume group snapshots.
+
+## How do I use Kubernetes Volume Group Snapshots
+
+Volume Group Snapshot feature is implemented in the external-snapshotter repo. It contains the following components:
+
+* Kubernetes Volume Group Snapshot CRDs
+* Volume group snapshot controller logic is added to the common snapshot controller.
+* Volume group snapshot validation webhook logic is added to the common snapshot validation webhook.
+* Logic to make CSI calls is added to CSI Snapshotter sidecar controller.
+
+The volume snapshot controller, CRDs, and validation webhook are deployed once per cluster, while the sidecar is bundled with each CSI driver.
+
+Therefore, it makes sense to deploy the volume snapshot controller, CRDs, and validation webhook as a cluster addon. It is strongly recommended that Kubernetes distributors bundle and deploy the volume snapshot controller, CRDs, and validation webhook as part of their Kubernetes cluster management process (independent of any CSI Driver).
+
+### Creating a new group snapshot with Kubernetes
+Once a VolumeGroupSnapshotClass object is defined and you have volumes you want to snapshot together, you may create a new group snapshot by creating a VolumeGroupSnapshot object.
+
+The source of the group snapshot specifies whether the underlying group snapshot should be dynamically created or if a pre-existing VolumeGroupSnapshotContent should be used. One of the following members in the source must be set.
+* Selector - Selector is a label query over persistent volume claims that are to be grouped together for snapshotting. This labelSelector will be used to match the label added to a PVC.
+* VolumeGroupSnapshotContentName - specifies the name of a pre-existing VolumeGroupSnapshotContent object representing an existing volume group snapshot.
+
+For dynamic provisioning, a selector must be set so that the snapshot controller can find PVCs with the matching labels to be snapshotted together.
+
+```
```

Comment:
You can mark this as YAML and get highlighting.

Reply:
Done